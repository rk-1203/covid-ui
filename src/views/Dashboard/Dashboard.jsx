import { useState, useEffect } from 'react';
import Button from '@mui/material/Button';

import Loader from '../../components/Loader/Loader';
import GlobalSummary from './GlobalSummary/GlobalSummary';
import Countries from './Countries/Countries';
import APIService from '../../services/api.service';
import "./Dashboard.scss";

function App() {

    const [statsSummary, setStatsSummary] = useState(null);
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        fetchCovidStats();
    }, []);
    
    const fetchCovidStats = () => {
        setIsLoading(true);
        APIService.fetchCovidStatsSummary().then(response => {
            setStatsSummary(response.data);
        }).finally(() => setIsLoading(false))
    }

    return (
        <div className="dashboard">
            {
                isLoading ? 
                    <Loader /> :
                    <>
                        <div className="refresh">
                            <Button variant="contained" onClick={fetchCovidStats}>Refresh</Button>
                        </div>
                        <GlobalSummary summary={statsSummary.Global}/>
                        <Countries countries={statsSummary.Countries}/>
                    </>
            }
        </div>
    );
}

export default App;
