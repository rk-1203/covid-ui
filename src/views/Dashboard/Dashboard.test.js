import React from 'react';
import { render, waitFor } from '@testing-library/react';

import Dashboard from "./Dashboard";
import APIService from "../../services/api.service";

jest.mock("../../services/api.service");

const fetchCovidStatsSummaryResponse = {
    ID: "10ab870f-bedb-427b-a27a-b54d87118b75",
    Message: "",
    Global: {
        NewConfirmed: 1929169,
        TotalConfirmed: 400178665,
        NewDeaths: 9389,
        TotalDeaths: 5760245,
        NewRecovered: 0,
        TotalRecovered: 0,
        Date: "2022-02-09T16:49:33.508Z"
    },
    Countries: [
        {
            ID: "e0296b5d-0c21-484d-be0f-7ee48a5f9a7b",
            Country: "Afghanistan",
            CountryCode: "AF",
            Slug: "afghanistan",
            NewConfirmed: 0,
            TotalConfirmed: 167739,
            NewDeaths: 0,
            TotalDeaths: 7454,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "ed082dda-b5ff-4d18-ad39-b97cc29e4de7",
            Country: "Barbados",
            CountryCode: "BB",
            Slug: "barbados",
            NewConfirmed: 0,
            TotalConfirmed: 49372,
            NewDeaths: 0,
            TotalDeaths: 288,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        }
    ],
    Date: "2022-02-09T16:49:33.508Z"
}

describe("Dashboard", () => {
    it("renders loading state properly", () => {
        APIService.fetchCovidStatsSummary.mockResolvedValue(new Promise(() => {}))
        const { container } = render(<Dashboard />);
        expect(container.firstChild).toMatchSnapshot();
    })

    it("renders loaded state properly", async () => {
        APIService.fetchCovidStatsSummary.mockResolvedValue(Promise.resolve({ data: fetchCovidStatsSummaryResponse}))
        const { container, getByText } = render(<Dashboard />);
        await waitFor(() => expect(APIService.fetchCovidStatsSummary).toBeCalledTimes(1));
        expect(container.firstChild).toMatchSnapshot();
    })
})