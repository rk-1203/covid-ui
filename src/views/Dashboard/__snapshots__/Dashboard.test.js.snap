// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders loaded state properly 1`] = `
<div
  class="dashboard"
>
  <div
    class="refresh"
  >
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Refresh
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <section
    class="global-summary"
  >
    <h1>
      Global Summary
    </h1>
    <div
      class="summary-items"
    >
      <div
        class="item"
      >
        <span
          class="label"
        >
          New Confirmed Cases
        </span>
        19,29,169
      </div>
      <div
        class="item"
      >
        <span
          class="label"
        >
          Total Confirmed Cases
        </span>
        40,01,78,665
      </div>
      <div
        class="item"
      >
        <span
          class="label"
        >
          New Deaths
        </span>
        9,389
      </div>
      <div
        class="item"
      >
        <span
          class="label"
        >
          Total Deaths
        </span>
        57,60,245
      </div>
      <div
        class="item"
      >
        <span
          class="label"
        >
          New Recovered Cases
        </span>
        0
      </div>
      <div
        class="item"
      >
        <span
          class="label"
        >
          Total Recovered Cases
        </span>
        0
      </div>
      <div
        class="item"
      >
        <span
          class="label"
        >
          Last Updated Time
        </span>
        09 Feb 2022, 10:19 PM
      </div>
    </div>
  </section>
  <div
    class="countries-container"
  >
    <div
      class="header"
    >
      <div
        class="count"
      >
        Total Countries : 
        2
      </div>
      <div
        class="filter"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="outlined-select-currency"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-select-currency"
              role="button"
              tabindex="0"
            >
              Country
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              placeholder="Search By"
              tabindex="-1"
              value="Country"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-nnbavb"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1lfthva-MuiInputBase-input-MuiFilledInput-input"
              id="standard-search"
              placeholder="Search"
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Country
            <span
              class="material-icons-outlined "
            >
              expand_less
            </span>
          </th>
          <th>
            New Confirmed
            <span
              class="material-icons-outlined unselected"
            >
              expand_less
            </span>
          </th>
          <th>
            Total Confirmed
            <span
              class="material-icons-outlined unselected"
            >
              expand_less
            </span>
          </th>
          <th>
            New Deaths
            <span
              class="material-icons-outlined unselected"
            >
              expand_less
            </span>
          </th>
          <th>
            Total Deaths
            <span
              class="material-icons-outlined unselected"
            >
              expand_less
            </span>
          </th>
          <th>
            New Recovered
            <span
              class="material-icons-outlined unselected"
            >
              expand_less
            </span>
          </th>
          <th>
            Total Recovered
            <span
              class="material-icons-outlined unselected"
            >
              expand_less
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="country-row"
        >
          <td
            data-th="Country"
          >
            Afghanistan
          </td>
          <td
            data-th="New Confirmed"
          >
            0
          </td>
          <td
            data-th="Total Confirmed"
          >
            1,67,739
          </td>
          <td
            data-th="New Deaths"
          >
            0
          </td>
          <td
            data-th="Total Deaths"
          >
            7,454
          </td>
          <td
            data-th="New Recovered"
          >
            0
          </td>
          <td
            data-th="Total Recovered"
          >
            0
          </td>
        </tr>
        <tr
          class="country-row"
        >
          <td
            data-th="Country"
          >
            Barbados
          </td>
          <td
            data-th="New Confirmed"
          >
            0
          </td>
          <td
            data-th="Total Confirmed"
          >
            49,372
          </td>
          <td
            data-th="New Deaths"
          >
            0
          </td>
          <td
            data-th="Total Deaths"
          >
            288
          </td>
          <td
            data-th="New Recovered"
          >
            0
          </td>
          <td
            data-th="Total Recovered"
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Dashboard renders loading state properly 1`] = `
<div
  class="dashboard"
>
  <div>
    <img
      alt="Loading..."
      height="200px"
      src="roundLoader.gif"
    />
  </div>
</div>
`;
